src/pages/index.js

```diff
@@ -0,0 +1,337 @@
+const addProfileModal = document.querySelector("#add-profile-modal");
+const addModalCloseBtn = addProfileModal.querySelector(".modal__close-btn");
+const addModalFormElement = addProfileModal.querySelector(".modal__form");
+const previewModal = document.querySelector("#preview-modal");
+const previewModalImage = previewModal.querySelector(".modal__image");
+const previewModalCaption = previewModal.querySelector(".modal__caption");
+
+const profileEditBtn = document.querySelector(".profile__edit-button");
+const profileTitle = document.querySelector(".profile__title");
+const profileDescription = document.querySelector(".profile__description");
+const profileAddBtn = document.querySelector(".profile__add-button");
+const cardTemplate = document.querySelector("#card-template");
+const cardsList = document.querySelector(".cards__list");
+const addCardLinkInput = addProfileModal.querySelector("#add-card-link-input");
+const addCardCaptionInput = addProfileModal.querySelector("#input-caption");
+const previewModalCloseBtn = previewModal.querySelector(".modal__close-btn");
+const addModalBtn = addProfileModal.querySelector(".modal__submit-btn");
+
+
+const deleteModal = document.querySelector("#delete-modal");
+const deleteModalClosebtn = deleteModal.querySelector(".modal__close-btn-delete");
+const deleteForm = deleteModal.querySelector(".delete__form");
+const deleteModalDeleteBtn = deleteModal.querySelector(".modal__submit-btn-delete");
+const deleteModalCancelBtn = deleteForm.querySelector(".modal__submit-btn-cancel");
+const deleteModalInput = deleteForm.querySelector(".modal__label-delete");
+
+const avatarModalBtn = document.querySelector(".profile__avatar-btn");
+const avatarModal = document.querySelector("#avatar-modal");
+const avatarSubmitBtn =  avatarModal.querySelector(".modal__submit-btn");
+const avatarCloseBtn = avatarModal.querySelector(".modal__close-btn");
+const avatarForm = avatarModal.querySelector(".modal__form");
+const avatarProfileInput = avatarModal.querySelector("#profile-avatar-input");
+
+let selectedCard , selectedCardId;
+
+function handleDeleteCard(cardElement,cardId){
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+
+function handleLike(evt,id){
+  const isLiked = evt.target.classList.contains("card__like-button_liked");
+  api.changeLikeStatus(id , isLiked)
+   .then(()=> {
+    evt.target.classList.toggle("card__like-button_liked"); })
+    .catch(console.error);
+  } ;
+
+function handleDeleteSubmit(evt){
+  evt.preventDefault();
+
+  const deleteModalDeleteBtn = evt.submitter;
+  setButtonText(deleteModalDeleteBtn,true,'Delete','Deleting...');
+
+  api.removeCard(selectedCardId)
+  .then(()=> {
+  selectedCard.remove();
+  closeModal(deleteModal);
+  setButtonText(deleteModalDeleteBtn,false,'Delete','Deleting...');
+  })
+  .catch((err) => {
+    console.error(err);
+    setButtonText(deleteModalDeleteBtn,false,'Delete','Deleting...');
```

Comment:
`  setButtonText(deleteModalDeleteBtn,false,'Delete','Deleting...');
`

It should be only in `.finally`, to avoid code duplication in `.then` and `.catch`